src/lib/protocols/connections/domain/did/__tests__/Service.test.ts

```diff
@@ -0,0 +1,148 @@
+import { classToPlain, plainToClass } from 'class-transformer';
+import { Service, ServiceTransformer, serviceTypes, IndyAgentService } from '../service';
+
+describe('Did | Service', () => {
+  it('should correctly transform Json to Service class', async () => {
+    const json = {
+      id: 'test-id',
+      type: 'Mediator',
+      serviceEndpoint: 'https://example.com',
+    };
+    const service = plainToClass(Service, json);
+
+    expect(service.id).toBe(json.id);
+    expect(service.type).toBe(json.type);
+    expect(service.serviceEndpoint).toBe(json.serviceEndpoint);
+  });
+
+  it('should correctly transform Service class to Json', async () => {
+    const json = {
+      id: 'test-id',
+      type: 'Mediator',
+      serviceEndpoint: 'https://example.com',
+    };
+
+    const service = new Service({
+      ...json,
+    });
+
+    const transformed = classToPlain(service);
+
+    expect(transformed).toEqual(json);
+  });
+
+  // TODO: make more generic like in PublicKey.test.ts
+  describe('IndyAgentService', () => {
+    it('should correctly transform Json to IndyAgentService class', async () => {
+      const json = {
+        id: 'test-id',
+        type: 'IndyAgent',
+        recipientKeys: ['917a109d-eae3-42bc-9436-b02426d3ce2c', '348d5200-0f8f-42cc-aad9-61e0d082a674'],
+        routingKeys: ['0094df0b-7b6d-4ebb-82de-234a621fb615'],
+        priority: 10,
+        serviceEndpoint: 'https://example.com',
+      };
+      const service = plainToClass(IndyAgentService, json);
+
+      expect(service.id).toBe(json.id);
+      expect(service.type).toBe(json.type);
+      expect(service.recipientKeys).toEqual(json.recipientKeys);
+      expect(service.routingKeys).toEqual(json.routingKeys);
+      expect(service.priority).toBe(json.priority);
+      expect(service.serviceEndpoint).toBe(json.serviceEndpoint);
```

Comment:
I guess we can't use `toMatchObject` because we try to test class attributes against JSON properties right?